plugins/auth-backend/src/providers/gitlab/provider.ts

```diff
@@ -47,29 +57,63 @@ type PrivateInfo = {
 
 export type GitlabAuthProviderOptions = OAuthProviderOptions & {
   baseUrl: string;
+  signInResolver?: SignInResolver<OAuthResult>;
+  authHandler: AuthHandler<OAuthResult>;
+  tokenIssuer: TokenIssuer;
+  catalogIdentityClient: CatalogIdentityClient;
+  logger: Logger;
 };
 
-function transformProfile(fullProfile: FullProfile) {
-  const profile = makeProfileInfo({
-    ...fullProfile,
-    photos: [
-      ...(fullProfile.photos ?? []),
-      ...(fullProfile.avatarUrl ? [{ value: fullProfile.avatarUrl }] : []),
-    ],
+export const extractGitLabUserId = (profile: ProfileInfo): string => {
+  return profile.email?.split('@')[0] as string;
+};
+
+export const gitlabDefaultSignInResolver: SignInResolver<OAuthResult> = async (
+  info,
+  ctx,
+) => {
+  const { result } = info;
+
+  const id = result.fullProfile.username || result.fullProfile.id;
```

Comment:
```suggestion
  let id = fullProfile.id;
  if (profile.email) {
    id = profile.email.split('@')[0];
  }
```

Just to be extra careful we wanna keep the existing logic here